### DISCUSSION TITLE
Inherit type parameters constraints in child classes

### DISCUSSION THREAD

**Discussion by tuscen** [dotnet/csharplang#6259] [Language Ideas] [answered]
Currently when you have a base generic class with constrained type parameters you need to restate all the constraints in every child class which is quite tedious and noisy. It would be much terser were you to have all constraints to be inheritable by child classes unless one needs to narrow a particular constraint even more.

Sorry if something like that has been proposed already, I couldn't find it.

## Example

Current behaviour:

```csharp
interface IFoo
{}

interface IBar : IFoo
{}

class A<T> where T : IFoo
{}

class B<T> : A<T> where T :IFoo
{}

class C<T> : A<T> where T : IBar
{}
```

Proposed behaviour:

```csharp
interface IFoo
{}

interface IBar : IFoo
{}

class A<T> where T : IFoo
{}

class B<T> : A<T> // where T : IFoo is implicitly inherited
{}

class C<T> : A<T> where T : IBar // the constraint IBar has to be stated explicitly since we're narrowing it down
{}
```

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/discussions/3810
